scene/gui/popup.cpp

```diff
@@ -108,11 +108,25 @@ void Popup::_close_pressed() {
 
 	_deinitialize_visible_parents();
 
-	call_deferred(SNAME("hide"));
+	// Hide after returning to process events, but only if we don't
+	// get popped up in the interim.
+	call_deferred(SNAME("_popup_conditional_hide"));
```

Comment:
I think you could remove the need for binding this method in `_bind_methods` (which makes it accessible from scripting, even if hidden by default due to the leading `_`) by using Callables instead of StringNames:

```suggestion
	callable_mp(this, &Popup::_popup_conditional_hide).call_deferred({}, 0);
```

CC @KoBeWi to confirm, currently we seem to only use `Callable::call_deferred` on a `callable_mp` once:
```
scene/3d/collision_object_3d.cpp
322:                    callable_mp(this, &CollisionObject3D::_update_debug_shapes).call_deferred({}, 0);
```

Reply:
Yes, thanks!  That's exactly what I wanted.  I will wait for confirmation.

Reply:
I have to admit though, if it is only used in one place right now, that seems a bit high on the risk/reward scale?  We already tested as written, including bruvzg's time.

Reply:
Well `Callable`s and `callable_mp` are new in 4.0 and while a big part of the codebase has been ported where it was trivial, I don't think we've really looked at deferred calls yet. But it should work fine, and that should be the "good" approach going forwards. The less we need to rely on strings, the better.

Reply:
As we don't agree on this, I'll do it myself in a follow up.

Reply:
No argument there.   I just wouldn't put that on this PR at this point in time.   I await your decision after discussion off-line.

Reply:
After further discussion, it was made clear that converting remaining uses of `call_deferred(StringName, VarArgs...)` is not that trivial and requires further discussion to define how we want to do this exactly (if at all).

Here's an example conversion in a case where there's actually an argument to pass to `call_deferred` and how it requires getting a `const Variant **` to pass:
```diff
diff --git a/modules/openxr/editor/openxr_action_map_editor.cpp b/modules/openxr/editor/openxr_action_map_editor.cpp
index 6e9a2e1b61..3d40e4853b 100644
--- a/modules/openxr/editor/openxr_action_map_editor.cpp
+++ b/modules/openxr/editor/openxr_action_map_editor.cpp
@@ -183,7 +183,8 @@ void OpenXRActionMapEditor::_on_add_action_set() {
 	// Make sure our action set is the current tab
 	tabs->set_current_tab(0);
 
-	call_deferred("_set_focus_on_action_set", new_action_set_editor);
+	const Variant *argp = &new_action_set_editor;
+	callable_mp(this, &OpenXRActionMapEditor::_set_focus_on_action_set).call_deferred(&argp, 1);
 }
 
 void OpenXRActionMapEditor::_set_focus_on_action_set(OpenXRActionSetEditor *p_action_set_editor) {
 ```
 
 Which is arguably more bothersome than the original code, even if it makes it possible to unbind the method.